src/Kriti.hs

```diff
@@ -38,6 +39,12 @@ runKriti templateSrc source = do
   template' <- first KritiParseError $ parser templateSrc'
   first KritiEvalError $ runEval templateSrc' template' source
 
+runKritiWith :: T.Text -> [(T.Text, J.Value)] -> [(T.Text, J.Value -> Either CustomFunctionError J.Value)] -> Either KritiError J.Value
```

Comment:
Could the functions come in as a Map rather than a list?

Reply:
Yeah sure, we are converting the list to a hashmap eventually. So, it makes sense to get a hashmap in the first place.